### ISSUE TITLE
Feature parity and ongoing support for `cloud` backend

### ISSUE THREAD

**Issue by robwittman** [opentofu/opentofu#2238]
### OpenTofu Version

```shell
1.9.0-alpha2
```

### The problem in your OpenTofu project

With 1.9.0 and further, opentofu released support for `-exclude` flags. Internally, tofu is still using `github.com/hashicorp/go-tfe`, which doesn't have support for this feature, and likely never will. As opentofu continues to develop features distinct from the hashicorp library, the gap in functionality supported by that library will continue to widen. For services to satisfy the `cloud` backend, the options they supported are currently limited by hashicorp, rather than whats provided by opentofu. 

### Attempted Solutions

I've personally forked `github.com/hashicorp/go-tfe`, and tested adding support for `-exclude` (and additionally, would do so for other features), but I don't imagine `opentofu` would want to rely on a 3rd party fork of that library. I can build my own fork of opentofu that uses the forked version of `go-tfe`, but that's a lot of workaround to support the `cloud` backend outside of upstream `opentofu` 

### Proposal

Less a proposal, and more a question? Do the maintainers of `opentofu` have any intention on supporting a fork of `github.com/hashicorp/go-tfe`, (or new project altogether) specific to the features and functionalities provided by `opentofu`?  Is there any desire to maintain support for the `cloud` backend in general? Looking at the functionality of the `cloud` backend, and its relationship with `go-tfe`, it would require the maintainers here to make opinionated decisions on the design of that API, structs, fields, etc. Maybe that makes sense in some way, maybe it doesn't. 

Maybe there's a longer path with some sort of backend "plugin" akin to providers; tofu prescribes events, their payload, and some sort of hooks or callbacks, and the backend providers can do the processing specific to their implementation 🤷 

TLDR; It would be great to see a fork of `go-tfe` for opentofu, built with features supported by it, and allowing ~100% feature parity for the `cloud` backend. I'm not sure what that would look like, if its even possible, or if it makes sense given the direction of the project, but I wanted to ask. I've been working with the cloud backend quite a bit, and I'd love to see the integrated nature of it continue

### References

https://github.com/opentofu/opentofu/blob/main/internal/cloud/backend_plan.go#L86 as an example, where we drop support for the `-exclude` feature on the cloud backend

**Comment by apparentlymart** [comment]
I've added this to the list of motivating issues for https://github.com/opentofu/opentofu/pull/2157, FWIW.

I'm not currently actively working on that since my attention is elsewhere, but I'd like to spend more time on it in the future since that area of things is long overdue for some attention.

**Comment by robwittman** [comment]
> We do not have access to the reference implementation for this protocol, nor is there a good and thorough documentation for it.

I've been hacking on an open source, self hosted project for supporting the `cloud` backend. While most of the API implementations are relatively complete, I'm still finalizing the runtime execution and log streaming. I'd be more than happy to see if there's any assistance I can help with on that front; either making it more accessible, documentation, or anything else. 

I saw some conversation in https://github.com/opentofu/opentofu/issues/960 around the documentation of those endpoints, expected payloads, and the like. I'd also be happy to start working on documentation on some of the internals, including the myriad of footguns. I've been working in that portion extensively the past year-ish, both on the open source project I mentioned, and professionally. If there's any direction or format folks wanted to see, just point me to it. 

> Many simply use the various TACOS as a kind of CI implementation and seem to be happy with it.

In a sense, this is a large part of my question, right? I used TFE extensively at a past job, and personally I think the integrated nature of it was great. But, there's a lot of people in the space solving similar problems without the `cloud` backend supported natively (at least, with all the features). I don't know that they're mutually exclusive, and if there were support for integrators to provide implementations for the cloud backend, or a similar interface, I think there's opportunity for good and _different_ solutions.
